api/v1beta1/awsloadbalancercontroller_types.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:validation:Enum=AWSShield;AWSWAFv1;AWSWAFv2
+type AWSAddon string
+
+const (
+	AWSAddonShield AWSAddon = "AWSShield"
+	AWSAddonWAFv1  AWSAddon = "AWSWAFv1"
+	AWSAddonWAFv2  AWSAddon = "AWSWAFv2"
+)
+
+// +kubebuilder:validation:Enum=Auto;Manual
+type SubnetTaggingPolicy string
+
+const (
+
+	// AutoSubnetTaggingPolicy enables automatic subnet tagging.
+	AutoSubnetTaggingPolicy SubnetTaggingPolicy = "Auto"
+
+	// ManualSubnetTaggingPolicy disables automatic subnet tagging.
+	ManualSubnetTaggingPolicy SubnetTaggingPolicy = "Manual"
+)
+
+// AWSLoadBalancerControllerSpec defines the desired state of AWSLoadBalancerController
+type AWSLoadBalancerControllerSpec struct {
```

Comment:
The description mentions `AWSLoadBalancerControllerSpec defines the desired state of AWSLoadBalancerController` even though `SubnetTagging` policy is related only to the operator. Using `AWSLoadBalancerSpec` would be closer to the intent. Though, given the type is already named as `awsloadbalancercontroller`, the renaming might create some confusion.

Reply:
I see what you mean. Yeah, the subnet tagging  belongs to all the instances of `AWSLoadBalancerController`. I don't remember we ever discussed the moving it away from the API because:
1) we are still sort of consistent because we can have only one`AWSLoadBalancerController` instance for the whole cluster (upstream limitation + we don't see a use case for it).
2) we liked the fact that we could use the status of the controller to clearly show the user which subnets are tagged/untagged/public/private. Moving this logic out of the API would make this reporting more difficult.

I think that we would have to revisit the subnet tagging if the need of having more than 1 instance of the controller comes up. However for now, a move like this may be a little too heavy to take especially given that we have the 17th of April as the deadline for the GA release of this operator.